catalog/transformer/catalog-transformer-streaming-impl/src/main/java/org/codice/ddf/transformer/xml/streaming/impl/XMLSaxEventHandlerImpl.java

```diff
@@ -0,0 +1,186 @@
+
+    }
+
+    @Override
+    public void endDocument() throws SAXException {
+
+    }
+
+    @Override
+    public void startPrefixMapping(String prefix, String uri) throws SAXException {
+
+    }
+
+    @Override
+    public void endPrefixMapping(String prefix) throws SAXException {
+
+    }
+
+    /**
+     * Takes in a sax event from {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}, and if it is
+     * in the xmlToMetacardMapping, begin reading it.
+     *
+     * @param uri        the URI that is passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     * @param localName  the localName that is passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     * @param qName      the qName that is passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     * @param attributes the attributes that are passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     */
+    @Override
+    public void startElement(String uri, String localName, String qName, Attributes attributes) {
+        /*
+         * If the element name is in the xmlToMetacardMapping, begin elementBeingRead. The data will be in the characters event.
+         */
+        if (xmlToMetacard.get(localName.toLowerCase()) != null) {
+            elementBeingRead = localName.toLowerCase();
+            return;
+        }
+
+        /*
+         * If the attribute "name" is in the xmlToMetacardMapping, take the "name" value, and add it directly to to the attributes list
+         */
+        String attribute = attributes.getValue("name");
+        if (attribute != null && xmlToMetacard.get(attribute) != null) {
+            elementBeingRead = attribute;
+        }
+    }
+
+    /**
+     * Takes in a sax event from {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}. If the element is in the xmlToElementMapping,
+     * add it to the attributes list
+     *
+     * @param namespaceURI the namespaceURI that is passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     * @param localName    the localName that is passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     * @param qName        the qName that is passed in by {@link org.codice.ddf.transformer.xml.streaming.lib.SaxEventHandlerDelegate}
+     */
+    @Override
+    public void endElement(String namespaceURI, String localName, String qName) {
+        /*
+         * If the handler is currently "reading" an element and the element is in the xmlToElementMapping, add the value of the
+         * textDataOfElement to the attributes list
+         */
+        if (elementBeingRead != null) {
+            String result = textDataOfElement.toString().trim();
+            textDataOfElement.setLength(0);
+            if (xmlToMetacard.get(elementBeingRead) != null) {
```

Comment:
Are we checking to see if the key exists in the hashmap, if the value for the key is null, or both?
If we are testing for the existence of the key, `xmlToMetacard.get(elementBeingRead) != null`
could be replaced with `xmlToMetacard.containsKey(elementBeingRead)`